### PR TITLE
[ENH] Abandon distutils

### LINKED ISSUE #1605: remove dependency on distutils
let's see if we can remove distutils dependencies 

```
DEPRECATION: Uninstalling a distutils installed project (nipype) has been deprecated 
and will be removed in a future version. This is due to the fact that uninstalling a 
distutils project will only partially uninstall the project.
```

### PR DESCRIPTION
**Pull request by oesteban**
This PR removes the dependency on ```distutils``` and uses only ```setuptools```.

Following [the recommendations](https://packaging.python.org/requirements/) I have reorganized the way nipype installs. Reading through [this discussion](https://www.reddit.com/r/Python/comments/3uzl2a/setuppy_requirementstxt_or_a_combination/) it seems that ```requirements.txt``` should be written for development environments, and production (pypi, mostly) dependencies should be handled with ```setuptools```.

Therefore, in this PR we start using all the available possibilities (```setup_requires```, ```install_requires```, ```test_requires```, ```extra_requires```, etc.) to install nipype smoothly.

I am testing all the possibilities with fresh environments:

- Regular user install through pypi, with the duecredit extra dependency:

  ```
  pip install nipype[duecredit]
  ```

- Regular user install from master:

  ```
  python setup.py install
  ```

- Developer install:

  ```
  pip install -r requirements.txt
  pip install -e .[test,doc]
  ```

- Testing install:

  ```
  pip install -r requirements.txt
  pip install -e .[all]
  ```

The available extra dependencies are listed here: https://github.com/oesteban/nipype/blob/enh/MigrateToSetuptools/nipype/info.py#L157

This PR requires #1623 to be merged first. Closes #1605

### CODE REVIEW

**Comment by oesteban** [comment]
One possible improvement to help novice users would be adding a check on the available third party tools (AFNI, FSL, FreeSurfer, etc.) and report it _ala_ matplotlib, warning the user about those missing software pieces.

**Comment by satra** [comment]
@oesteban - i would make that part of the new cli #1608 and it could internally use a function that checks and returns underlying tool status

**Comment by oesteban** [comment]
That sounds pretty cool!

Something like ```nipype check_deps``` ?

**Comment by satra** [comment]
@oesteban - resolve conflicts here.

File: nipype/info.py [hunk 1]
```diff
@@ -133,16 +132,38 @@ def get_nipype_gitversion():
...
+    "nose>=%s" % NOSE_MIN_VERSION,
+    "mock",
+    "codecov",
+    "doctest-ignore-unicode"
+]
+
+EXTRA_REQUIRES = {
+    'doc': ['Sphinx>=0.3'],
+    'tests': TESTS_REQUIRES,
+    'fmri': ['nitime', 'nilearn'],
```
Comment (satra) [inline nipype/info.py]: shouldn't this also have support for nipy and dipy and matplotlib?

File: nipype/info.py [hunk 2]
```diff
@@ -70,27 +70,26 @@ def get_nipype_gitversion():
 # Note: this long_description is actually a copy/paste from the top-level
 # README.txt, so that it shows up nicely on PyPI.  So please remember to edit
 # it only in one place and sync it correctly.
-long_description = \
-    """
+long_description = """\
```
Comment (satra) [inline nipype/info.py]: a triple-quoted string should not require a `\`

**Comment by satra** [comment]
@oesteban - should we merge this before or after the command line changes?

File: nipype/info.py (hunk 2 above)
Comment (oesteban) [inline nipype/info.py]: This is to avoid that newline, but I can remove it.

**Comment by oesteban** [comment]
@satra I'd do this before. When the command line can be added as an ```entry_point```, which is a better way of adding console scripts.

File: nipype/info.py (hunk 1 above)
Comment (oesteban) [inline nipype/info.py]: the fmri extra or the tests extras? I guess so, this is the best moment to decide the dependencies and how we organize them

File: nipype/info.py (hunk 2 above)
Comment (satra) [inline nipype/info.py]: You can make the string start right after the quotes.

File: nipype/info.py (hunk 1 above)
Comment (satra) [inline nipype/info.py]: perhaps for both, but at present for the various interfaces.

File: nipype/info.py (hunk 2 above)
Comment (oesteban) [inline nipype/info.py]: Ok, np. It just looked nicer to me this way. Probably not for others, will change XD

File: nipype/info.py (hunk 1 above)
Comment (oesteban) [inline nipype/info.py]: Ok!

Comment (oesteban) [inline nipype/info.py]: Actually, for the doc extra, what are the dependencies?? There were some graphviz-related dependencies here right?. Also matplotlib.

Comment (oesteban) [inline nipype/info.py]: Ok, found it. As per https://github.com/nipy/nipype/issues/1350, what do you think? pygraphwiz or pydot (or pydot2 https://github.com/nipy/nipype/issues/1280) - @satra

Comment (satra) [inline nipype/info.py]: all of them actually require graphviz installed. but if pydot2 or pydotplus works, i think those are the most recent things. i haven't looked into this in a while.

**Comment by oesteban** [comment]
Ok, this is ready. However, please double check that writing the COMMIT_INFO still works. Other than that, I think this is pretty done

**Comment by satra** [comment]
@oesteban - still has unicode issues:

on python 3, any stream from popen returns bytes and should be stripped and decoded.

```
In [2]: nipype.get_info()
Out[2]: 
{'commit_hash': "b'3e7dd7f\\n'",
 'commit_source': 'installation',
 'networkx_version': '1.11',
 'nibabel_version': '2.2.0dev',
 'numpy_version': '1.11.1',
 'pkg_path': '/software/anaconda/envs/dev3pype/lib/python3.5/site-packages/nipype-0.13.0_g3e7dd7f.dev-py3.5.egg/nipype',
 'scipy_version': '0.18.0',
 'sys_executable': '/software/anaconda/envs/dev3pype/bin/python',
 'sys_platform': 'darwin',
 'sys_version': '3.5.2 | packaged by conda-forge | (default, Jul 26 2016, 01:37:38) \n[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.54)]',
 'traits_version': '4.5.0'}
```

File: setup.py [hunk 3]
```diff
@@ -83,216 +59,39 @@ def get_comrec_build(pkg_dir, build_cmd=build_py):
...
+        import subprocess
+
+        build_py.run(self)
+        proc = subprocess.Popen('git rev-parse --short HEAD',
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE,
+                                shell=True)
+        repo_commit, _ = proc.communicate()
+        # Fix for python 3
+        repo_commit = '{}'.format(repo_commit)
```
Comment (satra) [inline setup.py]: this should use `repo_commit.decode()` on py3

**Comment by oesteban** [comment]
@satra - I've been also looking at getting rid of that long list of packages, using ```find_packages()``` from setuptools:

```
psetup = find_packages(exclude=['*.tests'])

print '\n'.join(list(set(packages)-set(psetup)))
nipype.testing.data.tbss_dir
nipype.caching.tests
nipype.interfaces.semtools.brains.tests
nipype.interfaces.semtools.tests
nipype.interfaces.elastix.tests
nipype.interfaces.mrtrix.tests
nipype.workflows.fmri.fsl.tests
nipype.interfaces.dipy.tests
nipype.utils.tests
nipype.interfaces.ants.tests
nipype.interfaces.slicer.legacy.diffusion.tests
nipype.interfaces.vista.tests
nipype.testing.data.bedpostxout
nipype.interfaces.semtools.segmentation.tests
nipype.interfaces.script_templates
nipype.interfaces.semtools.diffusion.tests
nipype.interfaces.diffusion_toolkit.tests
nipype.algorithms.tests
nipype.interfaces.slicer.legacy.tests
nipype.interfaces.slicer.diffusion.tests
nipype.interfaces.nitime.tests
nipype.pipeline.plugins.tests
nipype.workflows.dmri.fsl.tests
nipype.interfaces.slicer.segmentation.tests
nipype.interfaces.freesurfer.tests
nipype.interfaces.camino2trackvis.tests
nipype.interfaces.slicer.quantification.tests
nipype.interfaces.tests
nipype.interfaces.camino.tests
nipype.interfaces.fsl.tests
nipype.interfaces.cmtk.tests
nipype.interfaces.semtools.filtering.tests
nipype.interfaces.spm.tests
nipype.interfaces.mipav.tests
nipype.interfaces.mne.tests
nipype.interfaces.nipy.tests
nipype.interfaces.mrtrix3.tests
nipype.interfaces.semtools.registration.tests
nipype.interfaces.minc.tests
nipype.interfaces.slicer.registration.tests
nipype.interfaces.afni.tests
nipype.interfaces.semtools.diffusion.tractography.tests
nipype.interfaces.semtools.legacy.tests
nipype.pipeline.engine.tests
nipype.testing.data.dicomdir
nipype.testing.data
nipype.interfaces.semtools.utilities.tests
nipype.interfaces.slicer.filtering.tests
nipype.interfaces.slicer.tests
nipype.workflows.fmri.spm.tests
```

Do we really want to install all those ```*.tests``` packages? Then there are other folders that I'm not sure they should be packages as well (nipype.testing.data.tbss_dir, nipype.testing.data.bedpostxout, nipype.testing.data, nipype.testing.data.dicomdir, nipype.interfaces.script_templates, etc.)

File: setup.py (hunk 3 above)
Comment (oesteban) [inline setup.py]: Done

**Pull request merged**